### PR TITLE
Replace simple asset system with MultiCurrency

### PR DESCRIPTION
**Pull request by aliXsed**
The extended functionality of MultiCurrency in comparison with the simple asset interface will be useful in Cennzx benchmarking tests

### CODE REVIEW

File: crml/cennzx/src/lib.rs [hunk 1]
```diff
@@ -223,23 +238,23 @@ decl_module! {
 				Error::<T>::CannotAddLiquidityWithZero
 			);
 			ensure!(
-				T::AssetSystem::free_balance(core_asset_id, &from_account) >= core_amount,
+				T::MultiCurrency::free_balance(&from_account, Some(core_asset_id.clone())) >= core_amount,
```
Comment (jordy25519) [inline crml/cennzx/src/lib.rs]: nit: is asset_id copy? might not need all these clones

File: crml/cennzx/src/lib.rs [hunk 2]
```diff
@@ -73,9 +76,21 @@ pub trait Trait: frame_system::Trait {
...
+	/// The opaque token type for an imbalance of a particular currency. This is returned by unbalanced operations
+	/// and must be dealt with. It may be dropped but cannot be cloned.
+	type PositiveImbalance: Imbalance<Self::Balance, Opposite = Self::NegativeImbalance>;
+	/// Something which can provide multi currency asset management
+	type MultiCurrency: MultiCurrencyAccounting<
+		AccountId = Self::AccountId,
+		Balance = Self::Balance,
+		CurrencyId = Self::AssetId,
+		NegativeImbalance = Self::NegativeImbalance,
+		PositiveImbalance = Self::PositiveImbalance,
```
Comment (jordy25519) [inline crml/cennzx/src/lib.rs:92]: is it possible to leave these imbalance types unspecified and not on the module trait?

**Review by jordy25519 — APPROVED** [review]
lgtm!
few small nits
hoping we don't need to specify the imbalances type on the module

File: crml/cennzx/src/lib.rs (hunk 1 above)
Comment (aliXsed) [reply crml/cennzx/src/lib.rs]: Yes, it is. I will get rid of the clones.

File: crml/cennzx/src/lib.rs (hunk 2 above)
Comment (aliXsed) [reply crml/cennzx/src/lib.rs:92]: I tried it first without the imbalance types but got an error that Imbalance was not implemented as required by the MultiCurrencyAccounting::NegativeImbalance bound. The error was suggesting that I add a "where" clause to the cennzx Trait which I can try as an alternative solution.

File: crml/cennzx/src/lib.rs (hunk 1 above)
Comment (aliXsed) [reply crml/cennzx/src/lib.rs]: Done

File: crml/cennzx/src/lib.rs (hunk 2 above)
Comment (aliXsed) [reply crml/cennzx/src/lib.rs:92]: The "where" clause approach seems to create more problems.So leaving it as it is for this PR and we can get back to it later.

**Pull request merged**